libpromises/evalfunction.c

```diff
@@ -2202,7 +2212,7 @@ static FnCallResult FnCallGetIndicesClassic(EvalContext *ctx, ARG_UNUSED const P
 
 static FnCallResult FnCallGetIndices(EvalContext *ctx, ARG_UNUSED const Policy *policy, const FnCall *fp, const Rlist *finalargs)
 {
-    const char *name_str = RlistScalarValue(finalargs);
+    const char *name_str = RlistScalarValueSafe(finalargs);
```

Comment:
We will need to check for the type, yes, because calling `ResolveAndQualifyVarName()` with `name_str`, hence `[not printable]`, is definitely wrong. It may seem innocent, but it's a bug waiting to happen later.

But you're right, there are few places where this is a problem (only two that I can see, I'll mark the other one), so we only need to change it in those two places. The code change should not be that huge.